insights/parsers/horizon_local_settings_conf.py

```diff
@@ -0,0 +1,242 @@
+       line by line based on keys.
+
+       Attributes:
+           settings (dict): A dictionary containing all the configuration parameters and values.
+               Sample data::
+
+                   {
+                       'DEBUG': 'False',
+                       'LOGOUT_URL': '/dashboard/auth/logout/',
+                       'SECURE_PROXY_SSL_HEADER': ['HTTP_X_FORWARDED_PROTO', 'https'],
+                       'OPENSTACK_API_VERSIONS': {'identity': '3'},
+                       'OPENSTACK_KEYSTONE_MULTIDOMAIN_SUPPORT': 'True',
+                       'OPENSTACK_KEYSTONE_DEFAULT_DOMAIN': 'Default',
+                       'HORIZON_CONFIG["images_panel"]': 'legacy',
+                       'OPENSTACK_KEYSTONE_BACKEND': {
+                           'can_edit_domain': 'True',
+                           'can_edit_group': 'True',
+                           'can_edit_project': 'True',
+                           'can_edit_role': 'True',
+                           'can_edit_user': 'True',
+                           'name': 'native'
+                        },
+                        'API_RESULT_LIMIT': '1000',
+                        'POLICY_FILES_PATH': '/etc/openstack-dashboard',
+                        'REST_API_REQUIRED_SETTINGS': ['OPENSTACK_HYPERVISOR_FEATURES', 'LAUNCH_INSTANCE_DEFAULTS', 'OPENSTACK_IMAGE_FORMATS'],
+                        .......
+                   }
+    """
+
+    def parse_content(self, data):
+        """
+           Each line is searched for the first assignment operator to get the key
+           and then the values are extracted and converted to the corresponding data types.
+        """
+        # No data found or file is empty
+        if len(data) == 0:
+            raise ParseException("Empty file")
+
+        self.settings = {}
+        # Check if line does not starts with lowercase letters
+        local_settings = "\n".join([setting for setting in get_active_lines(data) if not re.match(r"^[a-z]+", setting)])
+
+        # Check if line contains a key
+        keys = []
+        for setting in local_settings.split("\n"):
+            key = re.findall(r"^[A-Z_\[\]\"]*\s*=", setting)
+            keys.extend(key) if len(key) > 0 else None
+
+        # No valid keys are found
+        if len(keys) == 0:
+            raise ParseException("No valid keys found.")
+
+        # Extract the keys and corresponding values from data and store in a dictionary
+        for key in range(len(keys)):
+            start = local_settings.index(keys[key]) + len(keys[key]) + 1
+            d_key = keys[key].strip("=").strip()
+            # Value is not present for a key
+            if (key < len(keys) - 1 and start == local_settings.index(keys[key + 1])) or \
+                    (key + 1 == len(keys) and not local_settings[start:]):
+                raise SkipException("Value not present for the key {0}.".format(d_key))
+            newline_index = local_settings[start:].find("\n")
```

Comment:
By adding the pair of "{" and "}", I think the following logic can be simplified by getting the first-right-searched "}" just before the next `key` to generate the value.

Reply:
And you should modify the code and make it easy to add handling for "()" and "[]" pairs.

Reply:
Acknowledged. Following changes have been done in the latest commit:
1. Remove unnecessary checks from the code
2. Extract data (value) present between current and next key
3. Handle Tuple and List by adding respective closing brackets